tests/v2/validation/pipeline/releaseupgrade/main.go

```diff
@@ -0,0 +1,395 @@
+		}
+	} else {
+		if isRKE1 {
+			switch cluster.Provider {
+			case provisioning.AWSProviderName.String():
+				config.AWSNodeTemplate.AMI = cluster.Image
+				config.ProvisioningConfig.Providers = []string{provisioning.AWSProviderName.String()}
+				config.AWSNodeTemplate.SSHUser = cluster.SSHUser
+			case provisioning.AzureProviderName.String():
+				config.AzureNodeTemplate.Image = cluster.Image
+				config.ProvisioningConfig.Providers = []string{provisioning.AzureProviderName.String()}
+				config.AzureNodeTemplate.SSHUser = cluster.SSHUser
+			case provisioning.DOProviderName.String():
+				config.DOMachineConfig.Image = cluster.Image
+				config.ProvisioningConfig.Providers = []string{provisioning.DOProviderName.String()}
+			case provisioning.HarvesterProviderName.String():
+				config.HarvesterNodeTemplate.ImageName = cluster.Image
+				config.ProvisioningConfig.Providers = []string{provisioning.HarvesterProviderName.String()}
+			case provisioning.LinodeProviderName.String():
+				config.LinodeNodeTemplate.Image = cluster.Image
+				config.ProvisioningConfig.Providers = []string{provisioning.LinodeProviderName.String()}
+			}
+		} else {
+			switch cluster.Provider {
+			case provisioning.AWSProviderName.String():
+				config.AWSMachineConfig.AMI = cluster.Image
+				config.ProvisioningConfig.Providers = []string{provisioning.AWSProviderName.String()}
+				config.AWSMachineConfig.SSHUser = cluster.SSHUser
+			case provisioning.AzureProviderName.String():
+				config.AzureMachineConfig.Image = cluster.Image
+				config.ProvisioningConfig.Providers = []string{provisioning.AzureProviderName.String()}
+				config.AzureMachineConfig.SSHUser = cluster.SSHUser
+			case provisioning.DOProviderName.String():
+				config.DOMachineConfig.Image = cluster.Image
+				config.ProvisioningConfig.Providers = []string{provisioning.DOProviderName.String()}
+				config.DOMachineConfig.SSHUser = cluster.SSHUser
+			case provisioning.HarvesterProviderName.String():
+				config.HarvesterMachineConfig.ImageName = cluster.Image
+				config.ProvisioningConfig.Providers = []string{provisioning.HarvesterProviderName.String()}
+				config.HarvesterMachineConfig.SSHUser = cluster.SSHUser
+			case provisioning.LinodeProviderName.String():
+				config.LinodeMachineConfig.Image = cluster.Image
+				config.ProvisioningConfig.Providers = []string{provisioning.LinodeProviderName.String()}
+				config.LinodeMachineConfig.SSHUser = cluster.SSHUser
+			}
+		}
+	}
+
+	config.UpgradeConfig = upgrade.Config{
+		Clusters: []upgrade.Clusters{
+			{
+				VersionToUpgrade: cluster.KubernetesVersionToUpgrade,
+				FeaturesToTest:   cluster.FeaturesToTest,
+			},
+		},
+	}
+}
+
+func wrapWithRunCommand(testCase string) string {
+	adminUserRegex := strings.ReplaceAll(provisioning.AdminClientName.String(), " ", "_")
+	return fmt.Sprintf(`-run \"%s/^%s\"`, testCase, adminUserRegex)
+}
+
+func parseYaml(fileName string, out interface{}) {
```

Comment:
Should parseYaml and updateYaml be extensions? They seem like they could be be broadly applicable.

Reply:
This function has been removed, we don't need this with the new changes.